core/controller/src/whisk/core/entitlement/Entitlement.scala

```diff
@@ -72,35 +78,33 @@ protected[core] object EntitlementService {
 /**
  * A trait for entitlement service. This is a WIP.
  */
-protected[core] abstract class EntitlementService(config: WhiskConfig)(implicit ec: ExecutionContext) extends Logging {
+protected[core] abstract class EntitlementService(config: WhiskConfig)(
+        implicit actorSystem: ActorSystem) extends Logging {
+
+    private implicit val executionContext = actorSystem.dispatcher
 
     private var loadbalancerOverload = false
     private val invokeRateThrottler = new RateThrottler(config, 120, 3600)
     private val triggerRateThrottler = new RateThrottler(config, 60, 720)
 
-    new Thread() {
-        /** query the KV store this often */
-        private val overloadCheckPeriodMillis = 10000
-
-        private val kvStore = new ConsulKV(config.consulServer)
-
-        /** Continously read from the KV store to get invoker status.*/
-        override def run() = {
-            var count = 0
-            while (true) {
-                val isOverload = kvStore.get(ConsulKV.LoadBalancerKeys.overloadKey)
-                isOverload match {
-                    case JsBoolean(v) => if ((count == 0) || loadbalancerOverload != v) {
-                        loadbalancerOverload = v
-                        info(this, s"EntitlementService: loadbalancerOverload = ${v}")
-                        count = count + 1
-                    }
-                    case _ => ()
-                }
-                Thread.sleep(overloadCheckPeriodMillis)
+    /** query the KV store this often */
+    private val overloadCheckPeriod = 10.seconds
+
+    private val kvClient = new ConsulClient(config.consulServer)
+
+    private var counted: Boolean = false
+    Scheduler.scheduleWaitAtLeast(overloadCheckPeriod) { () =>
+        kvClient.get(LoadBalancerKeys.overloadKey).map { isOverloaded =>
+            Try(isOverloaded.parseJson) match {
+                case Success(JsBoolean(v)) if !counted || loadbalancerOverload != v =>
```

Comment:
I tried to strictly follow what was there before; there was a `count` field being incremented and tested against `0`, so at least my version won't overflow :)